### PR TITLE
Clean up existing SK implementations and add a function calling counterpart

### PR DESCRIPTION
**Pull request by stephentoub**
- Change the 01 sample to just use InvokePromptAsync rather than using ChatHistory
- Add the 04-HikerAIPro counterpart that uses function calling
- Reduce boilerplate throughout
- Fix some grammar / typos / etc.

cc: @JakeRadMSFT 

(As we discussed offline, I don't fully understand how some of these samples are meant to differ, especially the second and third ones. The first was in that bucket as well, but my changes at least make it a little different.)

### CODE REVIEW

File: src/getting-started/infra/ai.bicep [hunk 1]
```diff
@@ -19,7 +19,7 @@ var openaiDeployment = [
     model: {
       format: 'OpenAI'
       name: 'gpt-35-turbo'
-      version: '0613'
+      version: '0125'
```
Comment (stephentoub) [inline src/getting-started/infra/ai.bicep:22]: I'm not sure what the best answer here is. 0613 is old and doesn't support tool calls, but 0914, 1106, and 0125 are available in fewer regions.

File: src/getting-started/semantic-kernel/01-HikeBenefitsSummary/Program.cs [hunk 2]
```diff
@@ -3,48 +3,29 @@
...
-chatHistory.Add(await chatCompletionService.GetChatMessageContentAsync(chatHistory, executionSettings));
-Console.WriteLine($"{chatHistory.Last().Role} >>> {chatHistory.Last().Content}");
+string endpoint = config["AZURE_OPENAI_ENDPOINT"];
+string deployment = config["AZURE_OPENAI_GPT_NAME"];
+string key = config["AZURE_OPENAI_KEY"];
+
+// Create a Kernel containing the Azure OpenAI Chat Completion Service
+Kernel kernel = Kernel.CreateBuilder()
+    .AddAzureOpenAIChatCompletion(deployment, endpoint, key)
+    .Build();
```
Comment (JakeRadMSFT) [inline src/getting-started/semantic-kernel/01-HikeBenefitsSummary/Program.cs:20]: How'd you choose between:

Kernel version:
`kernel.AddAzureOpenAIChatCompletion`

and

Using AOAI chat completion service directly?
`AzureOpenAIChatCompletionService service = new(deployment, endpoint, key);`

**Review by JakeRadMSFT — APPROVED** [review]

File: src/getting-started/semantic-kernel/01-HikeBenefitsSummary/Program.cs (hunk 2 above)
Comment (stephentoub) [reply src/getting-started/semantic-kernel/01-HikeBenefitsSummary/Program.cs:20]: I just wanted to use kernel.InvokePromptAsync.

File: src/getting-started/infra/ai.bicep (hunk 1 above)
Comment (JakeRadMSFT) [reply src/getting-started/infra/ai.bicep:22]: Hmm

https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#gpt-35-turbo-model-availability

Comment (JakeRadMSFT) [reply src/getting-started/infra/ai.bicep:22]: I think the best we can do is provide guidance some where in the readme/docs.

Comment (JakeRadMSFT) [reply src/getting-started/infra/ai.bicep:22]: https://github.com/dotnet/ai-samples/issues/28

**Pull request merged**